src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int MAX_SIZE = 100;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private Figure figure;
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(MathFigure.values().length);
+        String color = colorSupplier.getRandomColor();
+        switch (MathFigure.values()[index]) {
+            case SQUARE:
+                int side = random.nextInt(MAX_SIZE) + 1;
+                figure = new Square(color,side);
+                break;
+            case RECTANGLE:

```

Comment:
```suggestion
            case SQUARE:
                return getRandomSquare();
            case RECTANGLE:
```
lets add private methods to create a figure there
it would make your code much easier to read